### PR TITLE
refactor: convert errors.server.controller to typescript utility function

### PR DESCRIPTION
**Pull request by tshuli**
Reopens #327 
Part of #144

### CODE REVIEW

**Comment by tshuli** [comment]
@karrui thanks for your earlier [review](https://github.com/opengovsg/FormSG/pull/327), the comments were really helpful :)

I've made the edits for your re-review

File: src/app/utils/handle-mongo-error.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { MongoError } from 'mongodb'
+import { Error as MongooseError } from 'mongoose'
+
+// Default error message if no more specific error
+const defaultErrorMessage = 'An unexpected error happened. Please try again.'
```
Comment (karrui) [inline src/app/utils/handle-mongo-error.ts:5]: should this constant be in UPPER_SNAKE_CASE following convention?

File: src/app/utils/handle-mongo-error.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export const getMongoErrorMessage = (
+  err?: MongoError | MongooseError | string,
```
Comment (karrui) [inline src/app/utils/handle-mongo-error.ts:8]: Should we instead move `defaultErrorMessage` here so callers of this method can override the default with their own (for increased flexibility)?

```ts
export const getMongoErrorMessage = (
  err?: MongoError | MongooseError | string,
  defaultErrorMessage = 'An unexpected error happened. Please try again.'
```

Comment (karrui) [reply src/app/utils/handle-mongo-error.ts:8]: If doing this, please add test cases for it too

**Review by karrui — APPROVED** [review]
nice. Can you create an issue so we can use this utility to generate error strings when creating DatabaseErrors?

something like 
```ts
return errAsync(
  new DatabaseError(getMongoErrorMessage(error, 'optional default message if needed'))
)
```

**Pull request merged**